### DISCUSSION TITLE
State of OpenDataCam - Q3 2023

### DISCUSSION THREAD

**Discussion by vsaw** [opendatacam/opendatacam#603] [Announcements]
Hi everyone,

it has been a while since my last [State of OpenDataCam](https://github.com/opendatacam/opendatacam/discussions/565) post. So let's get everyone up to date what has happened since then.

I'll try to give you a broad overview over different areas of the Project including

- Maintainers & Developers
- Community
- Technology & Roadmap
- Outlook

If there is anything else you would like to discuss or have questions, please leave a comment.

## Maintainers & Developers

There has been a slight change in responsibilities in the core OpenDataCam team. While I was most active on GitHub, the OpenDataCam Domain and Accounts where still owned by @tdurand and previous members. All those accounts, domains and assets have now been transferred to me, which now makes me the main point of contact for all things OpenDataCam.

This change in ownership comes with good news: In the past few weeks, I've had many great conversations with members of the OpenDataCam Community and Companies looking forward to use OpenDataCam commercially. Those conversations clearly show how valuable OpenDataCam already is to many people and how huge the untapped potential of OpenDataCam still is. Therefore, **I've decided to work on OpenDataCam full time from now on!**. This means two things

1. OpenDataCam can now take on larger projects for commercial customers. In the past I had to say no to some larger requests since I was not able to do them while working my day job. So if you are a company looking forward to using OpenDataCam in your business reach out to me! (GitHub: @vsaw, Mastodon: https://mastodon.social/@vsaw, Email: [valentin@opendata.cam](mailto:valentin@opendata.cam))
1. Expect the project to make some larger leaps forward in the next weeks and months.

## Community

I'm happy to say that despite low activity, the OpenDataCam community remains stable.

![star-history-2023629](https://github.com/opendatacam/opendatacam/assets/829308/76ff28cc-4a3d-4f9d-bcd4-9d79e6753234)

OpenDataCam crossed the magic mark of 1500 [stargazers](https://github.com/opendatacam/opendatacam/stargazers) this month (congratulations @lehmannerich for being number 1500!). And, over the last 14 days. OpenDataCam has seen 709 Views and 289 Unique visitors on GitHub. This is roughly half then the last time I wrote a project update report, but given the fact that I did not have much chance to advocate for OpenDataCam in the past year, I'm still very positive about the continuing interest in the project.

To increase OpenDataCams visibility and growing the community I have planned three things

1. Restart development of OpenDataCam v4 (see part "Technology & Roadmap")
3. I will update the website of OpenDataCam in the near future with more useful information on what you can do with OpenDataCam, so that people who are not used to GitHub can still get a good impression of OpenDataCam.
1. Additionally, I've created [OpenDataCam on LinkedIn](https://www.linkedin.com/company/opendatacam) which I plan to use as another communication channel to make more people (read "potential customers that help me make a living from OpenDataCam") aware of OpenDataCam.

The OpenDataCam Twitter account [@opendatacam](https://twitter.com/opendatacam) I mentioned last time is currently on hold given all the turmoil the site went through. It still exists though and may be reactivated in the future if it makes sense.

## Technology & Roadmap

Basically, everything I said about the state of technology and development of OpenDataCam in [Q3 2022](https://github.com/opendatacam/opendatacam/discussions/565) is still true.

> From a technology perspective, OpenDataCam is falling behind on the State of the Art. The use of [Darknet](https://github.com/opendatacam/darknet) and Yolo v4 is keeping performance and detection results behind newer implementations such as [YOLOv7](https://github.com/WongKinYiu/yolov7) [*Update* in the meantime [YOLOv8](https://github.com/ultralytics/ultralytics) has been released]. On the Tracking side, [node-moving-things-tracker](https://github.com/opendatacam/node-moving-things-tracker) has severe issues tracking objects in busy scenes and can not resume tracks if objects become hidden (e.g. car driving behind a tree). All of these issues are solved in State of the Art trackers like [DeepSort](https://github.com/ZQPei/deep_sort_pytorch).

However these tackling those issues is in itself not trivial, and the fact that OpenDataCam is struggling with out of date dependencies makes it specially hard to tackle them.

Therefore OpenDataCam v4 will most likely be a maintenance release focusing on bringing OpenDataCam back on stable ground by

- making sure all documentation is up to date such as https://github.com/opendatacam/opendatacam/issues/503
- fixing all dependency issue such as https://github.com/opendatacam/opendatacam/issues/602
- adding support for current HW such as https://github.com/opendatacam/opendatacam/issues/405
- shipping important bug fixes such as https://github.com/opendatacam/opendatacam/issues/599
- introducing new features that are already working in development such as https://github.com/opendatacam/opendatacam/issues/187

The complete scope and progress of OpenDataCam v4 is managed via the public [Roadmap on GitHub](https://github.com/orgs/opendatacam/projects/2)

## Outlook

Given the fact that I will now be spending more time on OpenDataCam, I plan to restart the development of v4 and beyond, trying to catch up with the state of the art in Computer vision.

The reason why I will not yet communicate a date for future releases is because I will have to spend some of my time on paid work for OpenDataCam customers. I will have to prioritize these request over working on future open source releases, in order to being able to pay rent. But let me assure you, OpenDataCam will remain open-source and wherever possible I will use the funding to cross-finance work on the open source version of OpenDataCam!

For this I will keep reaching to different members of the community and potential clients in understanding how I could create a permanent paid offering that fits the OpenDataCam project. I will of course, keep being available for the OpenDataCam community in GitHub and answering as many questions as I can!

To help me to continue working full time on OpenDataCam, here are a few ideas what you can do:

- **Spread the word!** Recommend OpenDataCam to businesses and other people that are looking for a commercial solution.
- **Share your thoughts!** If you have an idea how OpenDataCam could generate revenue so that people like me can work on it full time, please reach out to me! (GitHub: @vsaw, Mastodon: https://mastodon.social/@vsaw, Email: [valentin@opendata.cam](mailto:valentin@opendata.cam))
- **Become a contributor!** There are many ways to contribute to OpenDataCam (Showcasing what you do with OpenDataCam, improving Documentation like setup instructions, or writing code) and all of them are welcome!

Thanks for making it this far and let's keep counting! ;-)

**Comment by tdurand** [comment]
🥳 to project revival !! Good luck and all the best !

**Reply by vsaw** [reply]
Thank you 🙏

**Comment by lehmannerich** [comment]
Thanks for the @mention 😄

On Thu 29. Jun 2023 at 17:10, Valentin ***@***.***> wrote:

> Hi everyone,
>
> it has been a while since my last State of OpenDataCam
> <https://github.com/opendatacam/opendatacam/discussions/565> post. So
> let's get everyone up to date what has happened since then.
>
> I'll try to give you a broad overview over different areas of the Project
> including
>
>    - Maintainers & Developers
>    - Community
>    - Technology & Roadmap
>    - Outlook
>
> If there is anything else you would like to discuss or have questions,
> please leave a comment.
> Maintainers & Developers
>
> There has been a slight change in responsibilities in the core OpenDataCam
> team. While I was most active on GitHub, the OpenDataCam Domain and
> Accounts where still owned by @tdurand <https://github.com/tdurand> and
> previous members. All those accounts, domains and assets have now been
> transferred to me, which now makes me the main point of contact for all
> things OpenDataCam.
>
> This change in ownership comes with good news: In the past few weeks, I've
> had many great conversations with members of the OpenDataCam Community and
> Companies looking forward to use OpenDataCam commercially. Those
> conversations clearly show how valuable OpenDataCam already is to many
> people and how huge the untapped potential of OpenDataCam still is.
> Therefore, *I've decided to work on OpenDataCam full time from now on!*.
> This means two things
>
>    1. OpenDataCam can now take on larger projects for commercial
>    customers. In the past I had to say no to some larger requests since I was
>    not able to do them while working my day job. So if you are a company
>    looking forward to using OpenDataCam in your business reach out to me!
>    (GitHub: @vsaw <https://github.com/vsaw>, Mastodon:
>    ***@***.***, Email: ***@***.***)
>    2. Expect the project to make some larger leaps forward in the next
>    weeks and months.
>
> Community
>
> I'm happy to say that despite low activity, the OpenDataCam community
> remains stable.
>
> [image: star-history-2023629]
> <https://user-images.githubusercontent.com/829308/249754186-76ff28cc-4a3d-4f9d-bcd4-9d79e6753234.png>
>
> OpenDataCam crossed the magic mark of 1500 stargazers
> <https://github.com/opendatacam/opendatacam/stargazers> this month
> (congratulations @lehmannerich <https://github.com/lehmannerich> for
> being number 1500!). And, over the last 14 days. OpenDataCam has seen 709
> Views and 289 Unique visitors on GitHub. This is roughly half then the last
> time I wrote a project update report, but given the fact that I did not
> have much chance to advocate for OpenDataCam in the past year, I'm still
> very positive about the continuing interest in the project.
>
> To increase OpenDataCams visibility and growing the community I have
> planned three things
>
>    1. Restart development of OpenDataCam v4 (see part "Technology &
>    Roadmap")
>    2. I will update the website of OpenDataCam in the near future with
>    more useful information on what you can do with OpenDataCam, so that people
>    who are not used to GitHub can still get a good impression of OpenDataCam.
>    3. Additionally, I've created OpenDataCam on LinkedIn
>    <https://www.linkedin.com/company/opendatacam> which I plan to use as
>    another communication channel to make more people (read "potential
>    customers that help me make a living from OpenDataCam") aware of
>    OpenDataCam.
>
> The OpenDataCam Twitter account @opendatacam
> <https://twitter.com/opendatacam> I mentioned last time is currently on
> hold given all the turmoil the site went through. It still exists though
> and may be reactivated in the future if it makes sense.
> Technology & Roadmap
>
> Basically, everything I said about the state of technology and development
> of OpenDataCam in Q3 2022
> <https://github.com/opendatacam/opendatacam/discussions/565> is still
> true.
>
> From a technology perspective, OpenDataCam is falling behind on the State
> of the Art. The use of Darknet <https://github.com/opendatacam/darknet>
> and Yolo v4 is keeping performance and detection results behind newer
> implementations such as YOLOv7 <https://github.com/WongKinYiu/yolov7> [
> *Update* in the meantime YOLOv8
> <https://github.com/ultralytics/ultralytics> has been released]. On the
> Tracking side, node-moving-things-tracker
> <https://github.com/opendatacam/node-moving-things-tracker> has severe
> issues tracking objects in busy scenes and can not resume tracks if objects
> become hidden (e.g. car driving behind a tree). All of these issues are
> solved in State of the Art trackers like DeepSort
> <https://github.com/ZQPei/deep_sort_pytorch>.
>
> However these tackling those issues is in itself not trivial, and the fact
> that OpenDataCam is struggling with out of date dependencies makes it
> specially hard to tackle them.
>
> Therefore OpenDataCam v4 will most likely be a maintenance release
> focusing on bringing OpenDataCam back on stable ground by
>
>    - making sure all documentation is up to date such as #503
>    <https://github.com/opendatacam/opendatacam/issues/503>
>    - fixing all dependency issue such as #602
>    <https://github.com/opendatacam/opendatacam/issues/602>
>    - adding support for current HW such as #405
>    <https://github.com/opendatacam/opendatacam/issues/405>
>    - shipping important bug fixes such as #599
>    <https://github.com/opendatacam/opendatacam/issues/599>
>    - introducing new features that are already working in development
>    such as #187 <https://github.com/opendatacam/opendatacam/issues/187>
>
> The complete scope and progress of OpenDataCam v4 is managed via the
> public Roadmap on GitHub <https://github.com/orgs/opendatacam/projects/2>
> Outlook
>
> Given the fact that I will now be spending more time on OpenDataCam, I
> plan to restart the development of v4 and beyond, trying to catch up with
> the state of the art in Computer vision.
>
> The reason why I will not yet communicate a date for future releases is
> because I will have to spend some of my time on paid work for OpenDataCam
> customers. I will have to prioritize these request over working on future
> open source releases, in order to being able to pay rent. But let me assure
> you, OpenDataCam will remain open-source and wherever possible I will use
> the funding to cross-finance work on the open source version of OpenDataCam!
>
> For this I will keep reaching to different members of the community and
> potential clients in understanding how I could create a permanent paid
> offering that fits the OpenDataCam project. I will of course, keep being
> available for the OpenDataCam community in GitHub and answering as many
> questions as I can!
>
> To help me to continue working full time on OpenDataCam, here are a few
> ideas what you can do:
>
>    - *Spread the word!* Recommend OpenDataCam to businesses and other
>    people that are looking for a commercial solution.
>    - *Share your thoughts!* If you have an idea how OpenDataCam could
>    generate revenue so that people like me can work on it full time, please
>    reach out to me! (GitHub: @vsaw <https://github.com/vsaw>, Mastodon:
>    ***@***.***, Email: ***@***.***)
>    - *Become a contributor!* There are many ways to contribute to
>    OpenDataCam (Showcasing what you do with OpenDataCam, improving
>    Documentation like setup instructions, or writing code) and all of them are
>    welcome!
>
> Thanks for making it this far and let's keep counting! ;-)
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/opendatacam/opendatacam/discussions/603>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABBUO4FA5UTAUKLUGOMZ6SLXNWLFDANCNFSM6AAAAAAZYU6OLI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
-----------------------------------------------------

Erich J. Lehmann
Co-Founder | MeetAnyway

www.meetanyway.com

Mob: +49 176 209 566 86
Email: ***@***.***

**Reply by vsaw** [reply]
Credit where credit is due 😀

**Comment by d3v3lop3rDE** [comment]
Thanks again for bringing this great project to my attention at @jax-mainz!

**Reply by vsaw** [reply]
Thank you for your kind words! 🙏
